std/utf.d

```diff
@@ -2729,7 +2729,8 @@ private P toUTFzImpl(P, S)(S str) @safe pure
     {
         typeof(*P.init)[] retval = ['\0'];
 
-        return retval.ptr;
+        auto trustedPtr() @trusted { return retval.ptr; }
+        return trustedPtr();
```

Comment:
Even though this LGTM, creating a function that can be called any time is more dangerous than an anonymous lambda that is called once:

```D
return (() @trusted => retval.ptr)();
```
And it's shorter code too.